app/src/main/java/com/kickstarter/libs/rx/transformers/NeverApiErrorTransformerV2.kt

```diff
@@ -0,0 +1,36 @@
+package com.kickstarter.libs.rx.transformers
+
+import com.kickstarter.services.apiresponses.ErrorEnvelope
+import com.kickstarter.services.apiresponses.ErrorEnvelope.Companion.fromThrowable
+import io.reactivex.Observable
+import io.reactivex.ObservableSource
+import io.reactivex.ObservableTransformer
+import io.reactivex.functions.Consumer
+
+open class NeverApiErrorTransformerV2<T> : ObservableTransformer<T, T> {
+    private val errorAction: Consumer<ErrorEnvelope>?
+
+    constructor() {
+        errorAction = null
+    }
+
+    protected constructor(errorAction: Consumer<ErrorEnvelope>?) {
+        this.errorAction = errorAction
+    }
+
+    override fun apply(upstream: Observable<T>): ObservableSource<T> {
+        return upstream
+            .doOnError { e: Throwable? ->
+                e?.let {
+                    val env = fromThrowable(it)
+                    env?.let { errorAction?.accept(env) }
+                }
+            }
+            .onErrorResumeNext { e: Throwable? ->
+                e?.let {
+                    val env = fromThrowable(it)
+                    env?.let { Observable.empty() } ?: Observable.error(e)
+                }
+            }
```

Comment:
@Arkariang Can you let me know if this too hard to understand? Is it more readable to do the regular old null checks like the original java code:
```
    return source
      .doOnError(e -> {
        final ErrorEnvelope env = ErrorEnvelope.fromThrowable(e);
        if (env != null && this.errorAction != null) {
          this.errorAction.call(env);
        }
      })
      .onErrorResumeNext(e -> {
        if (ErrorEnvelope.fromThrowable(e) == null) {
          return Observable.error(e);
        } else {
          return Observable.empty();
        }
      });
```